compiler/noirc_evaluator/src/ssa/checks/check_for_underconstrained_values.rs

```diff
@@ -433,4 +799,250 @@ mod test {
         let ssa_level_warnings = ssa.check_for_underconstrained_values();
         assert_eq!(ssa_level_warnings.len(), 1);
     }
+
+    #[test]
+    #[traced_test]
+    /// Test where a call to a Brillig function is left unchecked with a later assert,
+    /// by example of the program illustrating issue #5425 (simplified variant).
+    fn test_underconstrained_value_detector_5425() {
+        /*
+        unconstrained fn maximum_price(options: [u32; 2]) -> u32 {
+            let mut maximum_option = options[0];
+            if (options[1] > options[0]) {
+                maximum_option = options[1];
+            }
+            maximum_option
+        }
+
+        fn main(sandwiches: pub [u32; 2], drinks: pub [u32; 2], best_value: u32) {
+            let most_expensive_sandwich = maximum_price(sandwiches);
+            let mut sandwich_exists = false;
+            sandwich_exists |= (sandwiches[0] == most_expensive_sandwich);
+            sandwich_exists |= (sandwiches[1] == most_expensive_sandwich);
+            assert(sandwich_exists);
+
+            let most_expensive_drink = maximum_price(drinks);
+            assert(
+                best_value
+                == (most_expensive_sandwich + most_expensive_drink)
+            );
+        }
+
+        fn main f0 {
+          b0(v0: [u32; 2], v1: [u32; 2], v2: u32):
+            inc_rc v0
+            inc_rc v1
+            v4 = call f1(v0)
+            v6 = allocate
+            store u1 0 at v6
+            v7 = load v6
+            v11 = array_get v0, index u32 0
+            v12 = eq v11, v4
+            v13 = or v7, v12
+            store v13 at v6
+            v14 = load v6
+            v16 = array_get v0, index u32 1
+            v17 = eq v16, v4
+            v18 = or v14, v17
+            store v18 at v6
+            v19 = load v6
+            constrain v19 == u1 1
+            v22 = call f1(v1)
+            v23 = add v4, v22
+            v24 = eq v2, v23
+            constrain v2 == v23
+            dec_rc v0
+            dec_rc v1
+            return
+        }
+        */
+        let type_u32 = Type::Numeric(NumericType::Unsigned { bit_size: 32 });
+        let type_u1 = Type::Numeric(NumericType::Unsigned { bit_size: 1 });
+
+        let main_id = Id::test_new(0);
+        let mut builder = FunctionBuilder::new("main".into(), main_id);
```

Comment:
Please use the new SSA parser instead of the `FunctionBuilder` for these tests

Reply:
Where can I get a usage example?

Reply:
These are the current examples: https://github.com/search?q=repo%3Anoir-lang%2Fnoir%20ssa%3A%3Afrom_str&type=code

Reply:
Thank you! On it

Reply:
All the tests have been reworked, including the ones for the existing pass.